### DISCUSSION TITLE
O.P.Pick — issue with multiple nested picks from a single object

### DISCUSSION THREAD

**Discussion by mesqueeb** [millsp/ts-toolbelt#216] [Q&A] [answered]
#### Describe the bug

For this signature:
```ts
function pick<T extends Record<string, any>, K extends string> (obj: T, keys: F.AutoPath<T, K>[]): O.P.Pick<T, S.Split<K, '.'>>
```

this works:
<img width="640" alt="image" src="https://user-images.githubusercontent.com/3253920/110190161-43d8a400-7e65-11eb-92f8-3c897fcd3f99.png">

however, adding a second key breaks the nested pick:
<img width="635" alt="image" src="https://user-images.githubusercontent.com/3253920/110190181-5c48be80-7e65-11eb-8cfa-95bf5c998590.png">

#### Reproduce the bug
```ts
// REPL or a link to your repository if applicable.
// A *self-contained* demonstration of the problem.
```

Is it possible to use TS Toolbelt in the typescript TS Playground? If so it's easy for me to replicate this.
Otherwise I can set up a branch on my package if you like.

#### Possible Solution
I'm not sure!

**Comment by millsp** [comment]
Yes, it works in the playground. `AutoPath` cannot work on arrays like this, the inference won't work. Please, set a playground up and I'll look what I can do.

**Comment by millsp** [comment, marked as answer]
@mesqueeb I fixed the issue, sorry for moving this to discussions. I re-implemented to whole `O.P` family. Thanks for pointing this issue out. Let me know it it's all good! I added your use-case to the tests, I'm gonna do this for all user posted code from now on.

**Reply by mesqueeb** [reply]
Thank you so much!

**Comment by vatoutou** [comment]
Hey @mesqueeb, is your pick function return type right ?
I can only get Union of objects. And with `Union.Merge` I get an object but nested objects of this object are still Unions :/

**Reply by mesqueeb** [reply]
I think it's best you open a new ticket or discussion, and show your exact input, the current behavior and your expected behavior. : )
